features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -221,9 +221,9 @@ nsapi_error_t AT_CellularNetwork::set_registration(const char *plmn)
     } else {
         tr_debug("Manual network registration to %s", plmn);
         if (_op_act != RAT_UNKNOWN) {
-            return _at.at_cmd_discard("+COPS", "=1,2,", "%s%d", plmn, _op_act);
+            return _at.at_cmd_discard("+COPS", "=4,2,", "%s%d", plmn, _op_act);
         } else {
-            return _at.at_cmd_discard("+COPS", "=1,2", "%s", plmn);
+            return _at.at_cmd_discard("+COPS", "=4,2,", "%s", plmn);
```

Comment:
This changes network selection to be first manual and then automatic (mode=4). Do you have some specific reason for this change? Currently, if PLMN is defined only that PLMN is allowed for to avoid any unexpected costs, e.g. due to roaming.

Reply:
The purpose of having the auto fallback is to increase the chance of connecting to a network. I also noticed this is the setting used in AT command tester from M2Msupport. I wasn't aware that roaming cost was a factor. I can change it back to 1 if you think cost outweighs connectivity.

Reply:
Do you think COPS mode could be a configuration option in [mbed_lib.json](https://github.com/ARMmbed/mbed-os/blob/master/features/cellular/mbed_lib.json)?

Reply:
Not all modems support COPS mode 4, so that should also be defined for each modem as [CellularProperty](https://github.com/ARMmbed/mbed-os/blob/8d246d8ac10f2e8ee6c4dea68d3b594afbe206e7/features/cellular/framework/AT/AT_CellularBase.h#L50).

Reply:
I can add a config in mbed_lib.json to COPS mode. There're only two modes to select from:manual and manual then automatic. Do you have preference on the name of the config?
Yes, I can add COPS mode 4 to CellularProperty. Let me know if you have preference on the name of the enum.

Reply:
@linlingao It's your call :)

Reply:
I added a config in mbed_lib.json for choosing either manual or manual then automatic.
WRT adding a property in CellularProperty, I checked all the modems supported in our stack except RiotMicro RM1000 (I can't find their AT commands online), they all support mode 4. Do you still want me to add a property?

Reply:
BC95 supports just COPS <mode>=0, <mode>=1 and <mode>=2. That may have changed in some of the latest release?

Reply:
Most likely. According to their spec, 4 is supported.
https://www.quectel.com/UploadImage/Downlad/Quectel_BC95_AT_Commands_Manual_V1.9.pdf

Reply:
However, there is still a note that _Only <mode>=0, <mode>=1 and <mode>=2 are supported._ on the page 33.

Reply:
I missed the note. They shouldn't have listed 3 and 4 in parameters if they're not supported. Good catch.